examples/predict_on_grid_coordinates.py

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2017 The Verde Developers.
+# Distributed under the terms of the BSD 3-Clause License.
+# SPDX-License-Identifier: BSD-3-Clause
+#
+# This code is part of the Fatiando a Terra project (https://www.fatiando.org)
+#
+"""
+Gridding on a preexisting grid
+==============================
+
+The ``grid`` method of any Verde gridder allows us to interpolate the data on
+a regular grid. By passing the ``shape`` or ``spacing`` arguments, the method
+will build the regular grid by itself.
+Nevertheless, if we want to interpolate the data on a predefined regular grid,
+we can pass the grid ``coordinates`` instead.
+"""
+import matplotlib.pyplot as plt
+import cartopy.crs as ccrs
+import pyproj
+import numpy as np
+import verde as vd
+
+# Define a CheckerBoard gridder
+region = (0, 5000, -5000, 0)
+gridder = vd.datasets.CheckerBoard(region=region)
+
+# Define a set of grid coordinates as 1d arrays
+easting = np.linspace(*region[:2], 50)
+northing = np.linspace(*region[2:], 50)
+coordinates = (easting, northing)
```

Comment:
The example is supposed to show how to grid using an existing grid. Maybe best to make a grid and random points with the checkerboard, fit a spline to the points, and predict on the reference grid. That’s what we would use this feature for, right?

Reply:
As far as I remember, we agreed that the `grid` method will not handle the creation of the grid coordinates, just predict on the passed coords and handle the projection if any. In this way, in the future we would predict on a regular grid in two steps:
1. Create the grid coordinates with `grid_coordinates`
2. Use the `grid` method to predict on.

Something like:
```
longitude, latitude = vd.grid_coordinates(...)
grid = gridder.grid((longitude, latitude), projection=projection)
```

In this sense this is example is trying to show that behaviour. Does it make sense or should we change it?

Reply:
Yes, but maybe not at this point since we haven't made that change (and won't until 2.0). Right?

Reply:
Absolutely. What I'm asking is that this example should show this feature. On 2.0 we are going to deprecate the `spacing`, `shape`, etc arguments, but we can start instructing people to pass the grid coordinates to the `grid` method since 1.5. Right?

Reply:
Ah right, I guess so. We'd need one or more PRs making this consistent in the docs. But this example is called "Gridding on a pre-existing grid" so it's not showcasing the future syntax.

Reply:
You're right. So maybe we would need to change the pitch of the example, or maybe just remove it and update the docs on a different PR. What do you think?

Reply:
I'm fine with whichever you prefer. Since this is a still a bit tricky, it would be OK to not have a gallery example for this.